### ISSUE TITLE
Random Blank Page Issue

### ISSUE THREAD

**Issue by juliusgoh22** [wkhtmltopdf/wkhtmltopdf#4294]
**wkhtmltopdf version(s) affected**: 12.2.1

**OS information**
Ubuntu 16.04

**Description**
There is some random blank page throughout the whole pdf file.Attached screenshot is my problem. Is there any way to fix this kinda issue ?
![Screenshot 2019-03-20 at 3 51 12 PM](https://user-images.githubusercontent.com/34125609/54671473-f73d1680-4b30-11e9-9396-ee9ae09f3a53.png)

**Comment by copazze** [comment]
The problem is with your HTML table.
Try this : 
https://github.com/wkhtmltopdf/wkhtmltopdf/issues/3195
https://github.com/wkhtmltopdf/wkhtmltopdf/issues/3395

**Comment by juliusgoh22** [comment]
Yup that is what i did , page-break-before:always , thats why it will go like that , the content of my table is dynamic so i have no idea how to fix this , althought i fixed this but other place will have such issue too

**Comment by juliusgoh22** [comment]
if i add the into tr:last-child {page-break-inside: avoid !important} this is what will appear.
![Screenshot 2019-03-21 at 10 45 56 AM](https://user-images.githubusercontent.com/34125609/54730938-88120180-4bc6-11e9-8c02-dac7ca6f7559.png)

**Comment by PhilterPaper** [comment]
When you say "blank page", are you talking about the last bit of a table (even just the closing rules and border) spilling over onto the next page? That's what your example looks like to me. Sometimes controlling page-breaks with CSS to keep an entire row together works, but often it doesn't in wkHTMLtoPDF. It just doesn't do a good job of handling large tables.

If you have control over the HTML page, consider replacing the table by a series of `<div>`s with 0 margins top and bottom, that look like a table, but _might_ avoid being broken across pages. Then table headers and footers become a manual process (or at least, very complicated CSS), but maybe that will work for you.
